### PR TITLE
Use absolute addressing for parameters

### PR DESCRIPTION
**Pull request by klecki**
## Description:
Injecting full qualified path to parameters solve problems with lookup of enclosing entity and generating valid links for submodules that are runtime generated.

The missing-reference resolver of sphinx paramlinks assume that `source` of all nodes has format:
"/path/to/file.py:docstring of module.clsname.methname" and requires the `:`, while members of generated DALI submodules have only the:
"docstring of nvidia.dali.fn.decoders.image"
part, so the regex doesn't match. If the reference is fully specified, the plugin doesn't need to look it up and doesn't use the unsuitable regex.

Some nvidia.dali.ops module specific code is added, so we try to always obtain the `__call__` signature (it contains all inputs and kwargs) and we try to link to either class or `__call__` documentation depending if we are documenting input or kwarg - as that API has this specific distinction.

Other parts of API don't have specific conventions.

One example in pipeline doc showcases the ability to reference parameters of other functions.

### Affected modules and functionalities:
Sphinx doc generation

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [x] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [x] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

File: dali/python/nvidia/dali/pipeline.py [hunk 1]
```diff
@@ -147,10 +147,12 @@ class Pipeline(object):
         More details can be found in
         `this documentation section <advanced_topics_checkpointing.html>`_.
     py_num_workers : int, optional, default = 1
-        The number of Python workers that will process ``ExternalSource`` callbacks.
-        The pool starts only if there is at least one ExternalSource with ``parallel`` set to True.
+        The number of Python workers that will process :meth:`~nvidia.dali.fn.external_source`
```
Comment (mzient) [inline dali/python/nvidia/dali/pipeline.py:150]: While at it - I'd mention in this line that it's about _parallel_ external source callbacks.
```suggestion
        The number of Python workers that will process parallel :meth:`~nvidia.dali.fn.external_source`
```

**Comment by klecki** [comment]
!build

**Review by banasraf — APPROVED** [review]

**Pull request merged**